src/utils.js

```diff
@@ -5,3 +5,45 @@ export const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayo
 // useForceUpdate hook
 const forcedReducer = state => state + 1;
 export const useForceUpdate = () => useReducer(forcedReducer, 0)[1];
+
+export const createDeepProxy = (obj, affected) => {
+  const proxyMap = {};
+  const handler = {
+    get: (target, key) => {
+      if (!affected.has(target)) {
+        affected.set(target, [key]);
+      } else {
+        const used = affected.get(target);
+        if (!used.includes(key)) used.push(key);
+      }
+      const val = target[key];
+      if (typeof val !== 'object') {
+        return val;
+      }
+      if (!proxyMap[key]) {
```

Comment:
proxy map should be external to deepProxy to keep the same req equality between calls, or you will get always unique objects each render.

Reply:
Totally! I keep forgetting about the ref equality.

Reply:
And it is not trivial. In the end, I'd reinvent proxyEqual, which I really don't want to do.